### ISSUE TITLE
Hide maplibre logo

### ISSUE THREAD

**Issue by jay81979** [ramani-maps/ramani-maps#88]
I need to hide the maplibre logo. Is there currently a way to do this?

**Comment by JonasVautherin** [comment]
Is that allowed by the MapLibre/OpenStreetMap license? :thinking:

**Comment by JonasVautherin** [comment]
For OpenStreetMap: https://osmfoundation.org/wiki/Licence/Attribution_Guidelines#Interactive_maps. I don't know for MapLibre. So yeah it seems like it is allowed if you honor the attribution in one of the other mentioned ways.

**Comment by JonasVautherin** [comment]
Maybe it could be passed as a boolean to `MapLibre`, like `MapLibre(showAttribution: Boolean = true)`?

**Comment by klemensz** [comment]
It could be a property of the `org.ramani.compose.UiSettings`? But attribution is something different than the logo afaik (it's the small "info" button that usually opens some kind of dialog).

**Comment by JonasVautherin** [comment]
Makes sense in `UiSettings`! Let's call it `showLogo: Boolean`, which is consistent with the MapLibre docs. What do you think?

**Comment by jay81979** [comment]
Sounds awesome. I've just started working in Miplibre. I've got a lot of experience using Mapbox in compose but my new company prefers Maplibre. I've currently decided to use a fragment for Maplibre and include it in Compose with the AndroidFragment component but would like to move to your Compose wrapper when it's more mature.

I will keep working on my compose map using your compose wrapper in a seperate module and can contribute too.

**Comment by JonasVautherin** [comment]
> would like to move to your Compose wrapper when it's more mature.

What do you mean by "more mature"? Do you have something specific in mind?

**Comment by jay81979** [comment]
I'm currently converting a react native app with a complex Maplibre implementation into native Android. I'll let you know what I'm having trouble with implementing as I go. Reason for going native is there are performance issues to do with the react native bridge.

**Comment by jay81979** [comment]
I need to set compass, logo and attribution to be disabled for a start in the uisettings

**Comment by klemensz** [comment]
From my perspective, the properties of `org.maplibre.android.maps.UiSettings` are essential. They provide crucial control over various aspects of the map's UI, and most—if not all—are necessary for tailoring the user experience.

Here’s an example from a real-world app (MapBox, View-based):

```kt
uiSettings.isLogoEnabled = false
uiSettings.isAttributionEnabled = false
uiSettings.compassGravity = Gravity.TOP or Gravity.START
uiSettings.setCompassMargins(uiSettings.compassMarginLeft, uiSettings.compassMarginLeft * 4, 0, 0)
```

**Comment by JonasVautherin** [comment]
Would one of you be willing to open a PR exposing the necessary settings from MapLibre? That would be greatly appreciated 😊.

**Comment by klemensz** [comment]
How would you go about it? Do you have some ideas or directions?

**Comment by JonasVautherin** [comment]
I would try adding fields to `UiSettings` and see where it goes 🙈. I don't expect it to involve big changes, especially given that we already have a `UiSettings` argument 👍.

**Comment by JonasVautherin** [comment]
> converting them to data classes

Data classes should be avoided in the API: https://kotlinlang.org/docs/api-guidelines-backward-compatibility.html#avoid-using-data-classes-in-your-api. One goal is to have a stable API for Ramani eventually, so I thought it made sense to follow this advice :innocent:. What do you think?

**Comment by klemensz** [comment]
OK - please take a look at the PR https://github.com/ramani-maps/ramani-maps/pull/91 - I tried to keep the same structure and just extend it. 

Regarding API stability and binary compatibility, I'm not sure what will happen if the constructor is extended even if it's not a data class?

**Comment by annamatolay** [comment]
Do you happen to have any estimates for this? I'm asking because I'm working on a few days-long home assignment and am curious if I will be able to hide/move the logo before my deadline. As it conflicts with the MapLibre logo (and is also not required).

If there’s any way I can assist, please let me know! But I don't feel entitled to review a PR created for a library I just started to use today.

**Comment by JonasVautherin** [comment]
> But I don't feel entitled to review a PR created for a library I just started to use today.

Use @klemensz' branch (the one of PR #91) directly! So that you can hide the logo right away, and that's valuable testing for me: if it works for you, you can approve #91 and comment that you successfully used it to hide the logo 😊.

I will try it myself soon, but I do that in my free time so sometimes it takes a couple weeks before I get to do it.

In order to use @klemensz' branch in your project, use a gradle composite build. Gradle has documentation for it, but it should really just require you to:

1. Clone @klemensz' branch somewhere on your system, say in `/tmp/ramani-maps`
2. In your Android project, edit the `settings.gradle` (or `settings.gradle.kts`) and add `includeBuild("/tmp/ramani-maps/ramani-maplibre")`. Sync gradle in Android-Studio
3. That's it! Now your app should be using /tmp/ramani-maps/ramani-maplibre from sources already!

Let me know how it goes 🚀

**Comment by klemensz** [comment]
I tested my changes with the composite build approach you recommended, @JonasVautherin, and everything worked smoothly. While I only tested a few settings (not an exhaustive set), the key functionality appears to be stable.

I did need to slightly modify the `includeBuild` configuration to fit my setup, as I’m working in a project that still uses the Groovy DSL. Here’s the setup I used:

```groovy
includeBuild('/tmp/ramani-maps/ramani-maplibre') {
    dependencySubstitution {
        substitute module('org.ramani-maps:ramani-maplibre') using project(':')
    }
}
```

I also kept the standard dependency in `build.gradle`:

```groovy
implementation 'org.ramani-maps:ramani-maplibre:0.6.1'
```

This combination allowed the composite build to work correctly in my environment.
